<launch>
    <arg name="input_ultra_sonic_radar_topic" default="/sensing/ultra_sonic_radar/ultra_sonic_radar"/>
    <arg name="output_cloud" default="/sensing/ultra_sonic_radar/pointcloud" />
    <arg name="output_frame" default="base_link"/>
    <arg name="cloud_radius_m" default="0.2" />
    <arg name="cloud_resolution_m" default="0.01" />

    <node pkg="ultra_sonic_radar_detector" name="ultra_sonic_radar_detector_node" type="ultra_sonic_radar_detector_node">
        <param name="output_frame" value="$(arg output_frame)" />
        <param name="cloud_radius_m" value="$(arg cloud_radius_m)" />
        <param name="cloud_resolution_m" value="$(arg cloud_resolution_m)" />
        <remap from="~input/radar_0" to="$(arg input_ultra_sonic_radar_topic)_0"/>
        <remap from="~input/radar_1" to="$(arg input_ultra_sonic_radar_topic)_1"/>
        <remap from="~input/radar_2" to="$(arg input_ultra_sonic_radar_topic)_2"/>
        <remap from="~input/radar_3" to="$(arg input_ultra_sonic_radar_topic)_3"/>
        <remap from="~input/radar_4" to="$(arg input_ultra_sonic_radar_topic)_4"/>
        <remap from="~input/radar_5" to="$(arg input_ultra_sonic_radar_topic)_5"/>
        <remap from="~input/radar_6" to="$(arg input_ultra_sonic_radar_topic)_6"/>
        <remap from="~input/radar_7" to="$(arg input_ultra_sonic_radar_topic)_7"/>
        <remap from="~output/pointcloud" to="$(arg output_cloud)" />
    </node>
</launch>